# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 9dcbc290d52a
# Full Hash: 9dcbc290d52a141b932c336917aa339ed06c01f0
# Author: stransky <stransky@redhat.com>
# Date: 2022-08-03 21:28:30
# Description:
#   Bug 1777919 [Wayland] Don't crash on bogus move-to-rect callback r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D153580
# ==============================================================================

diff -r ba6a4bf96574 -r 9dcbc290d52a widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Wed Aug 03 12:03:55 2022 +0000
+++ b/widget/gtk/nsWindow.cpp	Wed Aug 03 12:09:01 2022 +0000
@@ -1859,11 +1859,16 @@
 
 void nsWindow::NativeMoveResizeWaylandPopupCallback(
     const GdkRectangle* aFinalSize, bool aFlippedX, bool aFlippedY) {
-#ifdef NIGHTLY_BUILD
   // We're getting move-to-rect callback without move-to-rect call.
-  // That indicates a compositor bug
-  MOZ_DIAGNOSTIC_ASSERT(mWaitingForMoveToRectCallback,
-                        "Bogus move-to-rect callback! A compositor bug?");
+  // That indicates a compositor bug. It happens when a window is hidden and
+  // shown again before move-to-rect callback is fired.
+  // It may lead to incorrect popup placement as we may call
+  // gtk_window_move() between hide & show.
+  // See Bug 1777919.
+#if MOZ_LOGGING
+  if (!mWaitingForMoveToRectCallback) {
+    LOG("  Bogus move-to-rect callback! A compositor bug?");
+  }
 #endif
 
   mWaitingForMoveToRectCallback = false;
